lib/src/parse/parser.dart

```diff
@@ -598,14 +603,17 @@ class Parser {
   }
 
   /// Consumes an identifier if its name exactly matches [text].
+  ///
+  /// [text] must only contain lowercase ASCII characters. It will be matched
```

Comment:
This restriction could either mean "any alphabetic ASCII characters in `text` must be lowercase" or "`text` may not contain any characters other than lowercase alphabetic ASCII characters". I think you mean the former, since it seems like it would make sense to write `scanIdentifier("else-if")`, but you should be explicit.

It's probably also not a bad idea to add an assertion that verifies this requirement as well.

Reply:
I'd like to make it possible for this to scan *any* identifier. I think it's mostly coincidental that we've managed to get away with only using it for alphabetical identifiers so far. Maybe have it check if the character is a letter and choose to do `scanLetter()` or `scanIdentChar()`, where the latter handles escape characters? Or else replace `scanLetter()` *with* `scanIdentChar()` and build in the case check there.

Reply:
Whoops. Missed this reply. Will do that